psyneulink/core/components/functions/function.py

```diff
@@ -327,6 +327,18 @@ def _output_type_setter(value, owning_component):
     return value
 
 
+def _seed_setter(value, owning_component, context):
```

Comment:
does this work with param modulation? would a change in modulated value reseed the random state?

Reply:
I don't think so. As it is now, the value of a parameter is only replaced with its corresponding parameter port's value when that port exists. I'm thinking of some changes for additional actions when getting a potentially modulated value. Assuming no changes to whatever does the modulating, these changes would be based on how often the reseed should happen. Using the example of an OCM running a simulation and changing the seed for each, do we want a reseed at the first time the `random_state` is requested per simulation/context?

Reply:
I think that for simulation purposes it only needs to happen once. Something like delaying random state initialization until the first request should work (we should see a modulated seed at that point).
It'd still be nice if the (modulated) seed that was used for initialization was reflected in the seed param.

Reply:
OK, I'll assume only automatically reseeding once per context. 

>It'd still be nice if the (modulated) seed that was used for initialization was reflected in the seed param.

This is sort of at odds with an intentional (but now old) decision that the main parameter value tracked/stored/got would be the base value, not modulated. The modulated value would be accessible through `Component.get_mod_seed` (autogenerated) or `Component._get_current_parameter_value`. I think it might be worth splitting this to its own issue, but we could add something to get the modulated value like `Parameter._get_modulated` or an additional argument to `Parameter._get` for example. I think keeping `_get` for just the base value makes sense.

Reply:
yeah, sounds reasonable. I think get_mod_seed does the job just fine.